packages/core/src/middleware/koa-error-log-handler.ts

```diff
@@ -0,0 +1,26 @@
+import { RequestErrorBody } from '@logto/schemas';
+import { Middleware } from 'koa';
+
+import { RequestErrorWithUserLog } from '@/errors/RequestError';
+import { insertUserLog } from '@/queries/user-log';
+
+export default function koaErrorLogHandler<StateT, ContextT>(): Middleware<
```

Comment:
just want to discuss, whether we should have a general `Log` handler that handles all kinds of logs `exceptions`, `infos`. 
Add put all the log db inserting logics at one place.  Where ever we call it, i.e. when catch a user flow errors or a general `sign-in` log, we simple call `log({type: 'user_log', payload,})`  so no need to call `insertUserLog` at multiple places.

Reply:
Since further discussion is needed, I'll ignore this problem in current PR, leave it to [LOG-395](https://linear.app/silverhand/issue/LOG-395/do-we-need-a-general-handler-for-inserting-log)

Reply:
[LOG-417](https://linear.app/silverhand/issue/LOG-417/do-we-need-a-general-handler-for-inserting-log), not LOG-395

Reply:
Refactored.